core/cthreads.c

```diff
@@ -0,0 +1,246 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "cthreads.h"
+
+#if _WIN32
```

Comment:
It might be better to use a slew of OS detection macros, on the off-chance someone is using some busted MinGW toolchain that didn't properly define _WIN32 (this is rare and shouldn't happen though)

Reply:
If you insist, I could use it, but I would probably not use in its official repository since I keep the usage of third party libraries at minimal. This isn't such a big problem since this is already a concord-specific version (includes only functions used by Concord. While I plan to add more, I won't be updating the cthreads.c for Concord, since it won't be used anyway, unless a bug fix)

I might be wrong, but Cygwin doesn't define `_WIN32` flag. I am not sure since I remember having to delete pthread part of CThreads to test Windows threads part

On the other side, pthread part is more tested, and it's preferable to be used, unless it could allow more bugs due to Cygwin pthread implementation